project/currency_comparison/index.js

```diff
@@ -0,0 +1,28 @@
+const readline = require("readline");
+const CurrencyComparison = require("./currency_comparison")
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+rl.question("Enter a yearly salary in USD: ", function(salary){
+    if (isNaN(salary)) {
+        console.log('Please try again with only enter in numeric characters.')
+        rl.close()
+    }  
+    const curSalary = new CurrencyComparison(salary)
+    rl.question("Enter a three-letter ISO currency code to compare with salary: ", async function(currency) {
+        const rates = await curSalary.fetchCurrentExchange()
+        if (!Object.keys(rates).includes(currency.toUpperCase())) {
+            console.log("Sorry, but it doesn't seem that we have currency exchange info for that currency.")
+            rl.close()
+        }
+            const exchangeRate = curSalary.currencyConversion(rates, currency.toUpperCase())
+            console.log(curSalary.response(currency.toUpperCase(), exchangeRate))
+            rl.close()
+    })
```

Comment:
What if we turned this logic into a function that:
1. accepts `currency` and a callback function
2. performs all of the logic above that relates to the `CurrencyComparison` instance but doesn't specify how to display the information (get rid of lines 17,18,21,22)
3. executes the callback with the `exchangeRate` value (or `null` if none was found for the given `currency`) instead

This would give us an async function that uses a callback to handle the response.